src/pstl_offload.cpp

```diff
@@ -61,7 +98,25 @@ __internal_free(void* __user_ptr)
         }
         else
         {
-            __get_original_free()(__user_ptr);
+            if (__dlsym_called)
+            {
+                // Delay releasing till exit of dlsym. We do not overload malloc globally,
+                // so can use it safely. Do not use new to able to use free() during
+                // __delayed_free_header releasing.
+                void* __buf = malloc(sizeof(__delayed_free_header));
+                if (!__buf)
+                {
+                    throw std::bad_alloc();
+                }
+                __delayed_free_header* __h = new(__buf) __delayed_free_header(__delayed_free, __user_ptr);
```

Comment:
```suggestion
                __delayed_free_list* __h = new(__buf) __delayed_free_list(__delayed_free, __user_ptr);
```

`delayed_free_header` doesn't reflect what this thing is doing at all.

Reply:
Done.